src/address.rs

```diff
@@ -120,10 +120,10 @@ impl Error for Disconnected {}
 pub struct Address<A, Rc: RefCounter = Strong> {
     pub(crate) sender: Sender<AddressMessage<A>>,
     pub(crate) ref_counter: Rc,
-    /// Activates once all contexts on this address have been dropped.
+    /// Resolves once all contexts on this address have been dropped.
     /// Used in [`Address::attach_stream`](struct.Address.html#method.attach_stream) to stop waiting
     /// for new messages when this `Address` becomes disconnected and in [`Address::join`].
-    pub(crate) tripwire: Tripwire
+    pub(crate) stop_notice: DropNotice,
```

Comment:
This will also lead to the address becoming bigger - do you think it'd be possible to store the event inside the DropNotice directly (i.e not behind an Arc), and then store the DropNotice inside the ref_counter's allocation, which is already an Arc?

One possible issue with this is that to wait on the stop notice, it would need to upgrade, which increases the strong count, so that alone would inhibit the actor being dropped. So, if that's done (which is probably better given that we want to keep the Address struct as small as it can be because there are usually a lot of them), we would need to use a separate counter for the number of actors, rather than reusing strong_count